src/cargo/core/registry.rs

```diff
@@ -166,9 +166,9 @@ impl<'cfg> PackageRegistry<'cfg> {
 
     pub fn register_lock(&mut self, id: PackageId, deps: Vec<PackageId>) {
         let sub_map = self.locked.entry(id.source_id().clone())
-                                 .or_insert(HashMap::new());
+                                 .or_insert_with(HashMap::new);
```

Comment:
I'd be interested to know whether the HashMap is always created in the old version, or if this can be optimized away.

(Maybe you can also to `.or_insert_with(|| HashMap::new())` if @alexcrichton just doesn't want to have this curried?)